Aligning Wallet Instance Attestation with Attestation-Based Client Authentication via HTTP Headers

cmarco0 opened the pull request:
this PR aims to close #289

cmarco0 committed:
replace http parameter to http header

cmarco0 committed:
replace http parameter to header

peppelinux commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -253,28 +253,35 @@ The PID/(Q)EAA Provider returns the issued ``request_uri`` to the Wallet Instanc
     Location: https://start.wallet.example.org?code=SplxlOBeZQQYbYS6WxSbIA&state=fyZiOL9Lf2CeKuNT2JzxiLRDink0uPcd&iss=https%3A%2F%2Fpid-provider.example.org
 
-**Steps 12-13 (DPoP Proof for Token Endpoint)**: The Wallet Instance MUST create a new key pair for the DPoP and a fresh DPoP Proof JWT following the instruction provided in Section 4 of (:rfc:`9449`) for the token request to the PID/(Q)EAA Provider. The DPoP Proof JWT is signed using the private key for DPoP created by Wallet Instance for this scope. DPoP binds the Access Token to a certain Wall ...
+**Steps 12-13 (DPoP Proof for Token Endpoint)**: The Wallet Instance MUST create a new key pair for the DPoP and a fresh DPoP Proof JWT following the instruction provided in this `specific <https://github.com/oauth-wg/draft-ietf-oauth-attestation-based-client-auth/blob/main/draft-ietf-oauth-attestation-based-client-auth.md>`_ for the token request to the PID/(Q)EAA Provider. The DPoP Proof JWT is ...
```
no, DPoP is not attestation based client auth

peppelinux commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -253,28 +253,35 @@ The PID/(Q)EAA Provider returns the issued ``request_uri`` to the Wallet Instanc
    2. It MUST ensure the Authorization ``code`` is valid and has not been previously used (:rfc:`6749`).
    3. It MUST ensure the ``redirect_uri`` matches the value included in the previous Request Object `OpenID.Core#TokenRequest <https://openid.net/specs/openid-connect-core-1_0.html#TokenRequest>`_.
-   4. It MUST validate the DPoP Proof JWT, according to (:rfc:`9449`) Section 4.3.
+   4. It MUST validate the DPoP Proof JWT, according to `OAuth 2.0 Attestation-Based Client Authentication <https://github.com/oauth-wg/draft-ietf-oauth-attestation-based-client-auth/blob/main/draft-ietf-oauth-attestation-based-client-auth.md>`_.
```
no

cmarco0 committed:
restore RFC reference in the text

peppelinux commented on docs/en/wallet-attestation.rst:
```diff
@@ -395,31 +398,31 @@ The body of the Wallet Attestation Request JWT MUST contain:
       -
     * - **cnf**
-      - JSON object, containing the public part of an asymmetric key pair owned by the Wallet Instance.
+      - MANDATORY. JSON object, containing the public part of an asymmetric key pair owned by the Wallet Instance.
```
```suggestion
      - REQUIRED. JSON object, containing the public part of an asymmetric key pair owned by the Wallet Instance.
```

peppelinux commented on docs/en/wallet-attestation.rst:
```diff
@@ -395,31 +398,31 @@ The body of the Wallet Attestation Request JWT MUST contain:
       - :rfc:`9126` and :rfc:`7519`.
     * - **iat**
-      - UNIX Timestamp with the time of JWT issuance.
+      - MANDATORY. UNIX Timestamp with the time of JWT issuance.
```
```suggestion
      - REQUIRED. UNIX Timestamp with the time of JWT issuance.
```

peppelinux commented on docs/en/wallet-attestation.rst:
```diff
@@ -395,31 +398,31 @@ The body of the Wallet Attestation Request JWT MUST contain:
       - :rfc:`9126` and :rfc:`7519`.
     * - **exp**
-      - UNIX Timestamp with the expiry time of the JWT.
+      - MANDATORY. UNIX Timestamp with the expiry time of the JWT.
```
```suggestion
      - REQUIRED. UNIX Timestamp with the expiry time of the JWT.
```

peppelinux commented on docs/en/wallet-attestation.rst:
```diff
@@ -395,31 +398,31 @@ The body of the Wallet Attestation Request JWT MUST contain:
       - **Reference**
     * - **iss**
-      - Identifier of the Wallet Provider concatenated with thumbprint of the JWK in the ``cnf`` parameter.
+      - MANDATORY. Identifier of the Wallet Provider concatenated with thumbprint of the JWK in the ``cnf`` parameter.
```
```suggestion
      - REQUIRED. Identifier of the Wallet Provider concatenated with thumbprint of the JWK in the ``cnf`` parameter.
```

cmarco0 committed:
editorial change; replace mandatory with required

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

cmarco0 committed:
editorial change; replace mandatory with required

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

cmarco0 committed:
editorial change; replace mandatory with required

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

cmarco0 committed:
editorial change; replace mandatory with required

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

cmarco0 committed:
Restore DPoP parameter in step 12-13

cmarco0 committed:
restore DPoP in steps 16-17

cmarco0 committed:
added oauth-client-attestation-pop

peppelinux commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -253,7 +253,7 @@ The PID/(Q)EAA Provider returns the issued ``request_uri`` to the Wallet Instanc
     Location: https://start.wallet.example.org?code=SplxlOBeZQQYbYS6WxSbIA&state=fyZiOL9Lf2CeKuNT2JzxiLRDink0uPcd&iss=https%3A%2F%2Fpid-provider.example.org
 
-**Steps 12-13 (DPoP Proof for Token Endpoint)**: The Wallet Instance MUST create a new key pair for the DPoP and a fresh DPoP Proof JWT following the instruction provided in Section 4 of (:rfc:`9449`) for the token request to the PID/(Q)EAA Provider. The DPoP Proof JWT is signed using the private key for DPoP created by Wallet Instance for this scope. DPoP binds the Access Token to a certain Wall ...
+**Steps 12-13 (DPoP Proof for Token Endpoint)**: The Wallet Instance MUST create a new key pair for the DPoP and a fresh DPoP Proof JWT following the instruction provided in this Section 4 of (:rfc:`9449`) for the token request to the PID/(Q)EAA Provider. The DPoP Proof JWT is signed using the private key for DPoP created by Wallet Instance for this scope. DPoP binds the Access Token to a certain ...
```
```suggestion
**Steps 12-13 (DPoP Proof for Token Endpoint)**: The Wallet Instance MUST create a new key pair for the DPoP and a fresh DPoP Proof JWT following the instruction provided in  the Section 4 of (:rfc:`9449`) for the token request to the PID/(Q)EAA Provider. The DPoP Proof JWT is signed using the private key for DPoP created by Wallet Instance for this scope. DPoP binds the Access Token to a certain Wallet Instance (:rfc:`9449`) and mitigates the misuse of leaked or stolen Access Tokens at the Credential Endpoint.
```

peppelinux commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -269,12 +269,20 @@ The ``client_assertion`` is signed using the private key that is created during
         oiUE9TVCIsImh0dSI6Imh0dHBzOi8vc2VydmVyLmV4YW1wbGUuY29tL3Rva2VuIiwia
         WF0IjoxNTYyMjYyNjE2fQ.2-GxA6T8lP4vfrg8v-FdWP0A0zdrj8igiMLvqRMUvwnQg
-        4PtFLbdLXiOSsX0x7NVY-FNyJK70nfbV37xRZT3Lg
+        4PtFLbdLXiOSsX0x7NVY-FNyJK70nfbV37xRZT3Lg    
```
```suggestion
        4PtFLbdLXiOSsX0x7NVY-FNyJK70nfbV37xRZT3Lg
```

peppelinux commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -536,16 +552,16 @@ The JOSE header of the Wallet Attestation proof of possession MUST contain:
+      - OPTIONAL. A digital signature algorithm identifier such as per IANA "JSON Web Signature and Encryption Algorithms" registry. It MUST be one of the supported algorithms listed in the Section `Cryptographic Algorithms <algorithms.html>`_ and MUST NOT be set to ``none`` or any symmetric algorithm (MAC) identifier.
       - :rfc:`7516#section-4.1.1`.
     * - **kid**
-      -  Unique identifier of the ``jwk`` inside the ``cnf`` claim of Wallet Attestation as base64url-encoded JWK Thumbprint value.
+      -  OPTIONAL. Unique identifier of the ``jwk`` inside the ``cnf`` claim of Wallet Attestation as base64url-encoded JWK Thumbprint value.
       - :rfc:`7638#section_3`.
```
@fmarino-ipzs it seems to me that kid to be references/contained in cnf.jwk doesn't have sense

may we remove this?

peppelinux commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -536,16 +552,16 @@ The JOSE header of the Wallet Attestation proof of possession MUST contain:
       - **Reference**
     * - **alg**
-      - A digital signature algorithm identifier such as per IANA "JSON Web Signature and Encryption Algorithms" registry. It MUST be one of the supported algorithms listed in the Section `Cryptographic Algorithms <algorithms.html>`_ and MUST NOT be set to ``none`` or any symmetric algorithm (MAC) identifier.
+      - OPTIONAL. A digital signature algorithm identifier such as per IANA "JSON Web Signature and Encryption Algorithms" registry. It MUST be one of the supported algorithms listed in the Section `Cryptographic Algorithms <algorithms.html>`_ and MUST NOT be set to ``none`` or any symmetric algorithm (MAC) identifier.
```
```suggestion
      - REQUIRED. A digital signature algorithm identifier such as per IANA "JSON Web Signature and Encryption Algorithms" registry. It MUST be one of the supported algorithms listed in the Section `Cryptographic Algorithms <algorithms.html>`_ and MUST NOT be set to ``none`` or any symmetric algorithm (MAC) identifier.
```

@cmarco0 why did you configure these as optional?

peppelinux commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -536,16 +552,16 @@ The JOSE header of the Wallet Attestation proof of possession MUST contain:
       - :rfc:`7638#section_3`.
     * - **typ**
-      -  It MUST be set to ``jwt-client-attestation-pop``
+      -  OPTIONAL. It MUST be set to ``jwt-client-attestation-pop``
```
```suggestion
      -  REQUIRED. It MUST be set to ``jwt-client-attestation-pop``
```

peppelinux commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -536,16 +552,16 @@ The JOSE header of the Wallet Attestation proof of possession MUST contain:
       -  Currently under discussion in [`oauth-attestation-draft <https://vcstuff.github.io/draft-ietf-oauth-attestation-based-client-auth/draft-ietf-oauth-attestation-based-client-auth.html>`_].
 
-The body of the Wallet Attestation proof of possession JWT MUST contain:
+The body of the Wallet Attestation proof of possession JWT contains:
```
@cmarco0 please don't change the requirements, the scope of this PR is to only change the HTTP Request POST parameters related to the attestation based client auth to the HTTP Request header parameters as defined in the latest draft.

peppelinux commented on docs/en/wallet-attestation.rst:
```diff
@@ -249,6 +249,7 @@ Below an non-normative example of the Wallet Attestation Request JWT without enc
       "jwk": {
         "crv": "P-256",
         "kty": "EC",
+        "use": "sig",
```
no, don't add nothing else to this PR.

peppelinux commented on docs/en/wallet-attestation.rst:
```diff
@@ -249,6 +249,7 @@ Below an non-normative example of the Wallet Attestation Request JWT without enc
+        "use": "sig",
         "x": "4HNptI-xr2pjyRJKGMnz4WmdnQD_uJSq4R95Nj98b44",
         "y": "LIZnSB39vFJhYgS3k7jXE4r3-CoGFQwZtPBIRqpNlrg",
         "kid": "vbeXJksM45xphtANnCiG6mCyuU4jfGNzopGuKvogg9c"
```
@fmarino-ipzs actually kid doesn't have sense in the cnf.jwk

peppelinux commented on docs/en/wallet-attestation.rst:
```diff
@@ -273,21 +274,27 @@ using the method `POST <https://datatracker.ietf.org/doc/html/rfc6749#section-3.
     POST /token HTTP/1.1
     Host: wallet-provider.example.org
     Content-Type: application/x-www-form-urlencoded
+    OAuth-Client-Attestation: eyJhbGciOiAiRVMyNTYiLCJraWQiOiAidmJlWEprc004NXhwaHRBTm5DaUc2bUN5dVU0amZHTnpvcEd1S3ZvZ2c5YyIsInR5cCI6ICJ3YXIrand0In0.eyJpc3MiOiAiaHR0cHM6Ly93YWxsZXQtcHJvdmlkZXIuZXhhbXBsZS5vcmcvaW5zdGFuY2UvdmJlWEprc004NXhwaHRBTm5DaUc2bUN5dVU0amZHTnpvcEd1S3ZvZ2c5YyIsInN1YiI6ICJodHRwczovL3dhbGxldC1wcm92aWRlci5leGFtcGxlLm9yZy8iLCJjaGFsbGVuZ2UiOiAiNmVjNjkzMjQtNjBhOC00ZTViLWE2OTctYTc2NmQ4N ...
```
remove `",`

peppelinux commented on docs/en/wallet-attestation.rst:
```diff
@@ -273,21 +274,27 @@ using the method `POST <https://datatracker.ietf.org/doc/html/rfc6749#section-3.
 **Steps 13-17**: The Wallet Provider's backend assesses the Wallet Attestation Request and issues a Wallet Attestation, if the requirements described below are satisfied:
 
-    1. It MUST check the Wallet Attestation Request contains all the defined parameters according to :ref:`Table of the Wallet Attestation Request parameters <table_wallet_attestation_request_claim>`.
+    1. It MUST check the Wallet Attestation Request contains all the defined header according to :ref:`Table of the Wallet Attestation Request Header <table_wallet_attestation_request_claim>`.
```
```suggestion
    1. It MUST check the Wallet Attestation Request contains all the defined HTTP Request header parameters according to :ref:`Table of the Wallet Attestation Request Header <table_wallet_attestation_request_claim>`.
```

peppelinux commented on docs/en/wallet-attestation.rst:
```diff
@@ -376,13 +383,13 @@ The JOSE header of the Wallet Attestation Request JWT MUST contain:
       - :rfc:`7638#section_3`.
     * - **typ**
-      -  It MUST be set to ``var+jwt``
+      -  OPTIONAL. It MUST be set to ``var+jwt``
```
it is not optional at all.

if it was MUST is therefore REQUIRED

cmarco0 committed:
editorial update. replace ; with .

cmarco0 committed:
editorial update. removed ", in the OAuth-Client-Attestation

cmarco0 committed:
editorial update. replace 'this' with 'the'

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

cmarco0 committed:
editorial update. removed space

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

cmarco0 committed:
editorial update. restore.

Restore requirements for JOSE header and body of Wallet Attestation

cmarco0 committed:
editorial update. restore.

restore requirements of JOSE header and body of Wallet Attestation

cmarco0 committed:
editorial update

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

cmarco0 committed:
editorial update. remove.

remove use claim from non-normative example of Wallet Attestation Request JWT without encoding and signature applied.

peppelinux commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -525,7 +541,7 @@ The JWT payload is given by the following parameters:
       - [:rfc:`7519`].
 
-The JOSE header of the Wallet Attestation proof of possession MUST contain:
+The JOSE header of the Wallet Attestation proof of possession MUST contains:
```
```suggestion
The JOSE header of the Wallet Attestation proof of possession MUST contain:
```

peppelinux commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -545,7 +561,7 @@ The JOSE header of the Wallet Attestation proof of possession MUST contain:
       -  Currently under discussion in [`oauth-attestation-draft <https://vcstuff.github.io/draft-ietf-oauth-attestation-based-client-auth/draft-ietf-oauth-attestation-based-client-auth.html>`_].
 
-The body of the Wallet Attestation proof of possession JWT MUST contain:
+The body of the Wallet Attestation proof of possession JWT MUST contains:
```
```suggestion
The body of the Wallet Attestation proof of possession JWT MUST contain:
```

peppelinux commented on docs/en/wallet-attestation.rst:
```diff
@@ -273,21 +273,27 @@ using the method `POST <https://datatracker.ietf.org/doc/html/rfc6749#section-3.
 The **token** endpoint (as defined in `RFC 7523 section 4`_) requires the following parameters
 encoded in ``application/x-www-form-urlencoded`` format:
 
+* ``OAuth-Client-Attestation`` A JWT that conforms to the structure and syntax as defined in the `OAuth 2.0 Attestation-Based Client Authentication` speficification;
```
```suggestion
* ``OAuth-Client-Attestation`` A JWT that conforms to the structure and syntax defined in the `OAuth 2.0 Attestation-Based Client Authentication` speficification;
```

peppelinux commented on docs/en/wallet-attestation.rst:
```diff
@@ -273,21 +273,27 @@ using the method `POST <https://datatracker.ietf.org/doc/html/rfc6749#section-3.
 encoded in ``application/x-www-form-urlencoded`` format:
 
+* ``OAuth-Client-Attestation`` A JWT that conforms to the structure and syntax as defined in the `OAuth 2.0 Attestation-Based Client Authentication` speficification;
+* ``OAuth-Client-Attestation-PoP`` A JWT that conforms to the structure and syntax as defined in the `OAuth 2.0 Attestation-Based Client Authentication` speficification;
```
```suggestion
* ``OAuth-Client-Attestation-PoP`` A JWT that conforms to the structure and syntax defined in the `OAuth 2.0 Attestation-Based Client Authentication` speficification;
```

peppelinux commented on docs/en/wallet-attestation.rst:
```diff
@@ -273,21 +273,27 @@ using the method `POST <https://datatracker.ietf.org/doc/html/rfc6749#section-3.
 * ``grant_type`` set to ``urn:ietf:params:oauth:grant-type:jwt-bearer``;
 * ``assertion`` containing the signed JWT of the Wallet Attestation Request.
 
+* ``OAuth-Client-Attestation and OAuth-Client-Attestation-PoP`` are both defined in this draft `specification <https://github.com/oauth-wg/draft-ietf-oauth-attestation-based-client-auth/blob/main/draft-ietf-oauth-attestation-based-client-auth.md>`_.
```
```suggestion
* ``OAuth-Client-Attestation and OAuth-Client-Attestation-PoP`` are both defined in `OAuth Attestation Based Client Authentication <https://github.com/oauth-wg/draft-ietf-oauth-attestation-based-client-auth/blob/main/draft-ietf-oauth-attestation-based-client-auth.md>`_.
```

cmarco0 committed:
editorial update.

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

cmarco0 committed:
editorial update

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

cmarco0 committed:
Editorial change, update hyperlink title

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

cmarco0 committed:
editorial change

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

cmarco0 committed:
editorial change

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

cmarco0 committed:
editorial update

fix oauth jwt.

the error was in the padding '=='

cmarco0 committed:
update to Oauth-client-attestation from client-attestation

cmarco0 committed:
update client_assertion to the latest draft OAuth 2.0 Attestation-Based Client

cmarco0 committed:
Merge branch 'versione-corrente' into cmarco0-AttestationBasedClientAuthenticationHTTPheaders

peppelinux commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -105,7 +105,7 @@ In this section a *Wallet Initiated Flow* is outlined, where the User receives t
   * MUST create the ``code_verifier`` with enough entropy random string using the unreserved characters with a minimum length of 43 characters and a maximum length of 128 characters, making it impractical for an attacker to guess its value. The value MUST be generated following the recommendation in Section 4.1 of :rfc:`7636`.
   * signs this request using the private key that is created during the setup phase to obtain the Wallet Attestation. The related public key that is attested by the Wallet Provider is provided within the Wallet Attestation ``cnf`` claim.
-  * MUST create the value of the ``client_assertion`` parameter according to OAuth 2.0 Attestation-based Client Authentication [`oauth-attestation-draft <https://vcstuff.github.io/draft-ietf-oauth-attestation-based-client-auth/draft-ietf-oauth-attestation-based-client-auth.html>`_], since in this flow the Pushed Authorization Endpoint is a protected endpoint. The ``client_assertion`` value MUST N ...
+  * MUST create the value of the ``OAuth-Client-Attestation`` and  ``OAuth-Client-Attestation-PoP`` parameter according to OAuth 2.0 Attestation-based Client Authentication [`oauth-attestation-draft <https://datatracker.ietf.org/doc/draft-ietf-oauth-attestation-based-client-auth/>`_], since in this flow the Pushed Authorization Endpoint is a protected endpoint. The first JWT is the Wallet Attesta ...
```
please use ref to defined specs and not URLs

peppelinux replied:
this url must be configured here: https://github.com/italia/eudi-wallet-it-docs/blob/versione-corrente/docs/common/standards.rst

and referenced in the text included in this PR

m-basili replied:
done

peppelinux replied:
therefore this url must be replaced with the ref you have defined

m-basili committed:
Update common_definitions.rst

m-basili committed:
Update standards.rst

added reference to specification "OAuth 2.0 Attestation-Based Client Authentication"

m-basili commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -105,7 +105,7 @@ In this section a *Wallet Initiated Flow* is outlined, where the User receives t
   * MUST create the ``code_verifier`` with enough entropy random string using the unreserved characters with a minimum length of 43 characters and a maximum length of 128 characters, making it impractical for an attacker to guess its value. The value MUST be generated following the recommendation in Section 4.1 of :rfc:`7636`.
   * signs this request using the private key that is created during the setup phase to obtain the Wallet Attestation. The related public key that is attested by the Wallet Provider is provided within the Wallet Attestation ``cnf`` claim.
-  * MUST create the value of the ``client_assertion`` parameter according to OAuth 2.0 Attestation-based Client Authentication [`oauth-attestation-draft <https://vcstuff.github.io/draft-ietf-oauth-attestation-based-client-auth/draft-ietf-oauth-attestation-based-client-auth.html>`_], since in this flow the Pushed Authorization Endpoint is a protected endpoint. The ``client_assertion`` value MUST N ...
+  * MUST create the value of the ``OAuth-Client-Attestation`` and  ``OAuth-Client-Attestation-PoP`` parameter according to OAuth 2.0 Attestation-based Client Authentication [`oauth-attestation-draft <https://datatracker.ietf.org/doc/draft-ietf-oauth-attestation-based-client-auth/>`_], since in this flow the Pushed Authorization Endpoint is a protected endpoint. The first JWT is the Wallet Attesta ...
```
```suggestion
  * MUST create the value of the ``OAuth-Client-Attestation`` and  ``OAuth-Client-Attestation-PoP`` parameter according to OAuth 2.0 Attestation-based Client Authentication [`OAUTH-ATTESTATION-CLIENT-AUTH`_ draft 03], since in this flow the Pushed Authorization Endpoint is a protected endpoint. The first JWT is the Wallet Attestation JWT and the second JWT is the Wallet Attestation Proof of Possession.
```

m-basili commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -120,7 +120,7 @@ The PID/(Q)EAA Provider performs the following checks upon the receipt of the PA
     9. It MUST check that the Request Object was issued in a previous time than the value exposed in the ``iat`` claim. It SHOULD reject the request if the ``iat`` claim is far from the current time (:rfc:`9126`) of more than `5` minutes.
     10. It MUST check that the ``jti`` claim in the Request Object has not been used before by the Wallet Instance identified by the ``client_id``. This allows the PID/(Q)EAA Provider to mitigate replay attacks (:rfc:`7519`).
-    11. It MUST validate the ``client_assertion`` parameter based on Sections 4.1 and 4.2 of [`oauth-attestation-draft <https://vcstuff.github.io/draft-ietf-oauth-attestation-based-client-auth/draft-ietf-oauth-attestation-based-client-auth.html>`_].
+    11. It MUST validate the ``OAuth-Client-Attestation-PoP`` parameter based on Section 4 of [`oauth-attestation-draft <https://datatracker.ietf.org/doc/draft-ietf-oauth-attestation-based-client-auth/>`_].
```
```suggestion
    11. It MUST validate the ``OAuth-Client-Attestation-PoP`` parameter based on Section 4 of [`OAUTH-ATTESTATION-CLIENT-AUTH`_ draft 03].
```

m-basili commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -348,7 +362,7 @@ Pushed Authorization Request Endpoint
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-The requests to the PID/(Q)EAA authorization endpoint MUST use the HTTP POST method with the parameters in the message body encoded in ``application/x-www-form-urlencoded`` format. The Pushed Authorization Endpoint is protected with OAuth 2.0 Attestation-based Client Authentication [`oauth-attestation-draft <https://vcstuff.github.io/draft-ietf-oauth-attestation-based-client-auth/draft-ietf-oauth ...
+The requests to the PID/(Q)EAA authorization endpoint MUST use the HTTP POST method with the parameters in the message body encoded in ``application/x-www-form-urlencoded`` format. The Pushed Authorization Endpoint is protected with OAuth 2.0 Attestation-based Client Authentication [`oauth-attestation-draft <https://datatracker.ietf.org/doc/draft-ietf-oauth-attestation-based-client-auth/>`_] and  ...
```
```suggestion
The requests to the PID/(Q)EAA authorization endpoint MUST use the HTTP POST method with the parameters in the message body encoded in ``application/x-www-form-urlencoded`` format. The Pushed Authorization Endpoint is protected with OAuth 2.0 Attestation-based Client Authentication [`OAUTH-ATTESTATION-CLIENT-AUTH`_ draft 03] and the following parameters MUST be provided:
```

m-basili commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -364,12 +378,12 @@ The requests to the PID/(Q)EAA authorization endpoint MUST use the HTTP POST met
-    * - **client_assertion**
-      - It MUST be set to a value containing the Wallet Attestation JWT and the Proof of Possession, separated with the ``~`` character.
-      - `oauth-attestation-draft <https://vcstuff.github.io/draft-ietf-oauth-attestation-based-client-auth/draft-ietf-oauth-attestation-based-client-auth.html>`_.
+    * - **OAuth-Client-Attestation**
+      - It MUST be set to a value containing the Wallet Attestation JWT.
+      - `oauth-attestation-draft <https://datatracker.ietf.org/doc/draft-ietf-oauth-attestation-based-client-auth/>`_.
+    * - **OAuth-Client-Attestation-PoP**
+      - It MUST be set to a value containing the Wallet Attestation JWT Proof of Possession.
+      - `oauth-attestation-draft <https://datatracker.ietf.org/doc/draft-ietf-oauth-attestation-based-client-auth/>`_.
```
```suggestion
    * - **OAuth-Client-Attestation**
      - It MUST be set to a value containing the Wallet Attestation JWT.
      - `OAUTH-ATTESTATION-CLIENT-AUTH`_ draft 03.
    * - **OAuth-Client-Attestation-PoP**
      - It MUST be set to a value containing the Wallet Attestation JWT Proof of Possession.
      - `OAUTH-ATTESTATION-CLIENT-AUTH`_ draft 03.
```

m-basili commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -598,14 +618,14 @@ Token endpoint
 
 The token endpoint is used by the Wallet Instance to obtain an Access Token by presenting an authorization grant, as
-defined in :rfc:`6749`. The Token Endpoint is a protected endpoint with a client authentication based on the model defined in OAuth 2.0 Attestation-based Client Authentication [`oauth-attestation-draft <https://vcstuff.github.io/draft-ietf-oauth-attestation-based-client-auth/draft-ietf-oauth-attestation-based-client-auth.html>`_].
+defined in :rfc:`6749`. The Token Endpoint is a protected endpoint with a client authentication based on the model defined in OAuth 2.0 Attestation-based Client Authentication [`oauth-attestation-draft <https://datatracker.ietf.org/doc/draft-ietf-oauth-attestation-based-client-auth/>`_].
```
```suggestion
defined in :rfc:`6749`. The Token Endpoint is a protected endpoint with a client authentication based on the model defined in OAuth 2.0 Attestation-based Client Authentication [`OAUTH-ATTESTATION-CLIENT-AUTH`_ draft 03].
```

m-basili commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -630,12 +650,12 @@ All the parameters listed below are REQUIRED:
-    * - **client_assertion**
-      - It MUST be set to a value containing the Wallet Attestation JWT and the Proof of Possession JWT, separated with the ``~`` character (WIA~WIA-PoP). The Wallet Attestation Proof of Possession MUST contain the claims as defined in :ref:`Table of the JWT Wallet Attestation PoP <table_jwt_pop>`, Section :ref:`Pushed Authorization Request (PAR) Response<sec_par>`.
-      - `oauth-attestation-draft <https://vcstuff.github.io/draft-ietf-oauth-attestation-based-client-auth/draft-ietf-oauth-attestation-based-client-auth.html>`_.
+    * - **OAuth-Client-Attestation**
+      - It MUST be set to a value containing the Wallet Attestation JWT.
+      - `oauth-attestation-draft <https://datatracker.ietf.org/doc/draft-ietf-oauth-attestation-based-client-auth/>`_.
+    * - **OAuth-Client-Attestation-PoP**
+      - It MUST be set to a value containing the Wallet Attestation JWT Proof of Possession JWT. The Wallet Attestation Proof of Possession MUST contain the claims as defined in :ref:`Table of the JWT Wallet Attestation PoP <table_jwt_pop>`, Section :ref:`Pushed Authorization Request (PAR) Response<sec_par>`.
+      - `oauth-attestation-draft <https://datatracker.ietf.org/doc/draft-ietf-oauth-attestation-based-client-auth/>`_.
```
```suggestion
    * - **OAuth-Client-Attestation**
      - It MUST be set to a value containing the Wallet Attestation JWT.
      - `OAUTH-ATTESTATION-CLIENT-AUTH`_ draft 03.
    * - **OAuth-Client-Attestation-PoP**
      - It MUST be set to a value containing the Wallet Attestation JWT Proof of Possession JWT. The Wallet Attestation Proof of Possession MUST contain the claims as defined in :ref:`Table of the JWT Wallet Attestation PoP <table_jwt_pop>`, Section :ref:`Pushed Authorization Request (PAR) Response<sec_par>`.
      - `OAUTH-ATTESTATION-CLIENT-AUTH`_ draft 03.
```

m-basili reviewed (commented):
Fixed reference link.

Commit:
Apply suggestions from code review

Co-authored-by: m-basili <m.basili@ipzs.it>

m-basili commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -364,12 +378,12 @@ The requests to the PID/(Q)EAA authorization endpoint MUST use the HTTP POST met
-    * - **client_assertion**
-      - It MUST be set to a value containing the Wallet Attestation JWT and the Proof of Possession, separated with the ``~`` character.
-      - `oauth-attestation-draft <https://vcstuff.github.io/draft-ietf-oauth-attestation-based-client-auth/draft-ietf-oauth-attestation-based-client-auth.html>`_.
+    * - **OAuth-Client-Attestation**
+      - It MUST be set to a value containing the Wallet Attestation JWT.
+      - `OAUTH-ATTESTATION-CLIENT-AUTH`_ draft 03.
+    * - **OAuth-Client-Attestation-PoP**
+      - It MUST be set to a value containing the Wallet Attestation JWT Proof of Possession.
+      - `OAUTH-ATTESTATION-CLIENT-AUTH`_ draft 03.
```
```suggestion
```

m-basili replied:
These parameters must be included in the HTTP header request and not in the payload of the request object.

peppelinux replied:
they must be moved in a separate table defining the http request header parameters

as @m-basili mentioned, in the current PR this table is only related to the http post body parameters and not http request header parameters

m-basili commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -444,6 +458,12 @@ The JWT payload is given by the following parameters:
     * - **redirect_uri**
       -  Redirection URI to which the response is intended to be sent. It MUST be an universal or app link registered with the local operating system, so this latter will provide the response to the Wallet Instance.
       - See `OpenID.Core#AuthRequest <https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest>`_.
+    * - **OAuth-Client-Attestation**
+      - It MUST be set as in the :ref:`Table of the HTTP parameters <table_http_request_claim>`.
+      - See :ref:`Table of the HTTP parameters <table_http_request_claim>`.
+    * - **OAuth-Client-Attestation-PoP**
+      - It MUST be set as in the :ref:`Table of the HTTP parameters <table_http_request_claim>`.
+      - See :ref:`Table of the HTTP parameters <table_http_request_claim>`.
```
```suggestion
```

peppelinux replied:
see my previous comment, it confirms @m-basili 's review and asks a specific table about the http request header parameters to be included within the request

m-basili commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -598,14 +618,14 @@ Token endpoint
 ^^^^^^^^^^^^^^^
 
-The request to the PID/(Q)EAA Token endpoint MUST be an HTTP request with method POST, with the body message encoded in ``application/x-www-form-urlencoded`` format. The Wallet Instance sends the Token endpoint request with ``client_assertion_type=urn:ietf:params:oauth:client-assertion-type:jwt-client-attestation`` and ``client_assertion=WIA~WIA-PoP``.
+The request to the PID/(Q)EAA Token endpoint MUST be an HTTP request with method POST, with the body message encoded in ``application/x-www-form-urlencoded`` format. The Wallet Instance sends the Token endpoint request with ``OAuth-Client-Attestation`` and ``OAuth-Client-Attestation-PoP``.
```
```suggestion
The request to the PID/(Q)EAA Token endpoint MUST be an HTTP request with method POST, with the body message encoded in ``application/x-www-form-urlencoded`` format. The Wallet Instance sends the Token endpoint request with ``OAuth-Client-Attestation`` and ``OAuth-Client-Attestation-PoP`` as header parameters according to `OAUTH-ATTESTATION-CLIENT-AUTH`_.
```

peppelinux replied:
remove draft 03

m-basili commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -630,12 +650,12 @@ All the parameters listed below are REQUIRED:
-    * - **client_assertion**
-      - It MUST be set to a value containing the Wallet Attestation JWT and the Proof of Possession JWT, separated with the ``~`` character (WIA~WIA-PoP). The Wallet Attestation Proof of Possession MUST contain the claims as defined in :ref:`Table of the JWT Wallet Attestation PoP <table_jwt_pop>`, Section :ref:`Pushed Authorization Request (PAR) Response<sec_par>`.
-      - `oauth-attestation-draft <https://vcstuff.github.io/draft-ietf-oauth-attestation-based-client-auth/draft-ietf-oauth-attestation-based-client-auth.html>`_.
+    * - **OAuth-Client-Attestation**
+      - It MUST be set to a value containing the Wallet Attestation JWT.
+      - `OAUTH-ATTESTATION-CLIENT-AUTH`_ draft 03.
+    * - **OAuth-Client-Attestation-PoP**
+      - It MUST be set to a value containing the Wallet Attestation JWT Proof of Possession JWT. The Wallet Attestation Proof of Possession MUST contain the claims as defined in :ref:`Table of the JWT Wallet Attestation PoP <table_jwt_pop>`, Section :ref:`Pushed Authorization Request (PAR) Response<sec_par>`.
+      - `OAUTH-ATTESTATION-CLIENT-AUTH`_ draft 03.
```
```suggestion
```

m-basili commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -299,6 +305,14 @@ A non-normative example of the Credential Request is provided below.
   Content-Type: application/json
   Authorization: DPoP Kz~8mXK1EalYznwH-LC-1fBAo.4Ljp~zsPE_NeO.gxU
   DPoP: eyJ0eXAiOiJkcG9wK2p3dCIsImFsZyI6IkVTMjU2IiwiandrIjp7Imt0eSI6Ik
+  OAuth-Client-Attestation: eyJhbGciOiAiRVMyNTYiLCJraWQiOiAiMTEifQ.eyJ
+        pc3MiOiJodHRwczovL2NsaWVudC5leGFtcGxlLmNvbSIsInN1YiI6Imh0dHBzOi8vY2x
+        pZW50LmV4YW1wbGUuY29tIiwibmJmIjoxMzAwODE1NzgwLCJleHAiOjEzMDA4MTkzODA
+        sImNuZiI6eyJqd2siOnsia3R5IjoiRUMiLCJ1c2UiOiJzaWciLCJjcnYiOiJQLTI1NiI
+        sIngiOiIxOHdITGVJZ1c5d1ZONlZEMVR4Z3BxeTJMc3pZa01mNko4bmpWQWlidmhNIiw
+        ieSI6Ii1WNGRTNFVhTE1nUF80Zlk0ajhpcjdjbDFUWGxGZEFnY3g1NW83VGtjU0EifX1
+        9.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c
+  OAuth-Client-Attestation-PoP: eyJ0eXAiOiJkcG9wK2p3dCIsImFsZyI6IkVTMjU2IiwiandrIjp7Imt0eSI6Ik
```
```suggestion
```

m-basili replied:
The Credential Endpoint is not protected by Client Authentication.

m-basili reviewed (changes requested):
some changes are required by commit suggestion in accordace with https://datatracker.ietf.org/doc/draft-ietf-oauth-attestation-based-client-auth/03/

Commit:
Merge branch 'versione-corrente' into cmarco0-AttestationBasedClientAuthenticationHTTPheaders

peppelinux commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -598,14 +618,14 @@ Token endpoint
 
 The token endpoint is used by the Wallet Instance to obtain an Access Token by presenting an authorization grant, as
-defined in :rfc:`6749`. The Token Endpoint is a protected endpoint with a client authentication based on the model defined in OAuth 2.0 Attestation-based Client Authentication [`oauth-attestation-draft <https://vcstuff.github.io/draft-ietf-oauth-attestation-based-client-auth/draft-ietf-oauth-attestation-based-client-auth.html>`_].
+defined in :rfc:`6749`. The Token Endpoint is a protected endpoint with a client authentication based on the model defined in OAuth 2.0 Attestation-based Client Authentication [`OAUTH-ATTESTATION-CLIENT-AUTH`_ draft 03].
```
```suggestion
defined in :rfc:`6749`. The Token Endpoint is a protected endpoint with a client authentication based on the model defined in OAuth 2.0 Attestation-based Client Authentication [`OAUTH-ATTESTATION-CLIENT-AUTH`_ ].
```

I suggest to keep draft version out from the text and only defining the draft url in the specs href pointed by the ref

peppelinux commented on docs/en/wallet-attestation.rst:
```diff
@@ -293,21 +293,27 @@ using the method `POST <https://datatracker.ietf.org/doc/html/rfc6749#section-3.
 * ``grant_type`` set to ``urn:ietf:params:oauth:grant-type:jwt-bearer``;
 * ``assertion`` containing the signed JWT of the Wallet Attestation Request.
 
+* ``OAuth-Client-Attestation and OAuth-Client-Attestation-PoP`` are both defined in `OAuth Attestation Based Client Authentication <https://github.com/oauth-wg/draft-ietf-oauth-attestation-based-client-auth/blob/main/draft-ietf-oauth-attestation-based-client-auth.md>`_.
```
don't use `<https://github.com/oauth-wg/draft-ietf-oauth-attestation-based-client-auth/blob/main/draft-ietf-oauth-attestation-based-client-auth.md>` but a ref to the defined specs

peppelinux commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -105,7 +105,7 @@ In this section a *Wallet Initiated Flow* is outlined, where the User receives t
   * MUST create the ``code_verifier`` with enough entropy random string using the unreserved characters with a minimum length of 43 characters and a maximum length of 128 characters, making it impractical for an attacker to guess its value. The value MUST be generated following the recommendation in Section 4.1 of :rfc:`7636`.
   * signs this request using the private key that is created during the setup phase to obtain the Wallet Attestation. The related public key that is attested by the Wallet Provider is provided within the Wallet Attestation ``cnf`` claim.
-  * MUST create the value of the ``client_assertion`` parameter according to OAuth 2.0 Attestation-based Client Authentication [`oauth-attestation-draft <https://vcstuff.github.io/draft-ietf-oauth-attestation-based-client-auth/draft-ietf-oauth-attestation-based-client-auth.html>`_], since in this flow the Pushed Authorization Endpoint is a protected endpoint. The ``client_assertion`` value MUST N ...
+  * MUST create the value of the ``OAuth-Client-Attestation`` and  ``OAuth-Client-Attestation-PoP`` parameter according to OAuth 2.0 Attestation-based Client Authentication [`OAUTH-ATTESTATION-CLIENT-AUTH`_ draft 03], since in this flow the Pushed Authorization Endpoint is a protected endpoint. The first JWT is the Wallet Attestation JWT and the second JWT is the Wallet Attestation Proof of Posse ...
```
```suggestion
  * MUST create the value of the ``OAuth-Client-Attestation`` and  ``OAuth-Client-Attestation-PoP`` parameter according to OAuth 2.0 Attestation-based Client Authentication [`OAUTH-ATTESTATION-CLIENT-AUTH`_], since in this flow the Pushed Authorization Endpoint is a protected endpoint. The first JWT is the Wallet Attestation JWT and the second JWT is the Wallet Attestation Proof of Possession.
```

peppelinux commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -120,7 +120,7 @@ The PID/(Q)EAA Provider performs the following checks upon the receipt of the PA
     9. It MUST check that the Request Object was issued in a previous time than the value exposed in the ``iat`` claim. It SHOULD reject the request if the ``iat`` claim is far from the current time (:rfc:`9126`) of more than `5` minutes.
     10. It MUST check that the ``jti`` claim in the Request Object has not been used before by the Wallet Instance identified by the ``client_id``. This allows the PID/(Q)EAA Provider to mitigate replay attacks (:rfc:`7519`).
-    11. It MUST validate the ``client_assertion`` parameter based on Sections 4.1 and 4.2 of [`oauth-attestation-draft <https://vcstuff.github.io/draft-ietf-oauth-attestation-based-client-auth/draft-ietf-oauth-attestation-based-client-auth.html>`_].
+    11. It MUST validate the ``OAuth-Client-Attestation-PoP`` parameter based on Section 4 of [`OAUTH-ATTESTATION-CLIENT-AUTH`_ draft 03].
```
```suggestion
    11. It MUST validate the ``OAuth-Client-Attestation-PoP`` parameter based on Section 4 of [`OAUTH-ATTESTATION-CLIENT-AUTH`_].
```

peppelinux commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -348,7 +362,7 @@ Pushed Authorization Request Endpoint
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-The requests to the PID/(Q)EAA authorization endpoint MUST use the HTTP POST method with the parameters in the message body encoded in ``application/x-www-form-urlencoded`` format. The Pushed Authorization Endpoint is protected with OAuth 2.0 Attestation-based Client Authentication [`oauth-attestation-draft <https://vcstuff.github.io/draft-ietf-oauth-attestation-based-client-auth/draft-ietf-oauth ...
+The requests to the PID/(Q)EAA authorization endpoint MUST use the HTTP POST method with the parameters in the message body encoded in ``application/x-www-form-urlencoded`` format. The Pushed Authorization Endpoint is protected with OAuth 2.0 Attestation-based Client Authentication [`OAUTH-ATTESTATION-CLIENT-AUTH`_ draft 03] and the following parameters MUST be provided:
```
```suggestion
The requests to the PID/(Q)EAA authorization endpoint MUST use the HTTP POST method with the parameters in the message body encoded in ``application/x-www-form-urlencoded`` format. The Pushed Authorization Endpoint is protected with OAuth 2.0 Attestation-based Client Authentication [`OAUTH-ATTESTATION-CLIENT-AUTH`_] and the following parameters MUST be provided:
```

peppelinux commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -364,12 +378,12 @@ The requests to the PID/(Q)EAA authorization endpoint MUST use the HTTP POST met
-      - `oauth-attestation-draft <https://vcstuff.github.io/draft-ietf-oauth-attestation-based-client-auth/draft-ietf-oauth-attestation-based-client-auth.html>`_.
+    * - **OAuth-Client-Attestation**
+      - It MUST be set to a value containing the Wallet Attestation JWT.
+      - `OAUTH-ATTESTATION-CLIENT-AUTH`_ draft 03.
```
```suggestion
      - `OAUTH-ATTESTATION-CLIENT-AUTH`_.
```

peppelinux commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -364,12 +378,12 @@ The requests to the PID/(Q)EAA authorization endpoint MUST use the HTTP POST met
+      - `OAUTH-ATTESTATION-CLIENT-AUTH`_ draft 03.
+    * - **OAuth-Client-Attestation-PoP**
+      - It MUST be set to a value containing the Wallet Attestation JWT Proof of Possession.
+      - `OAUTH-ATTESTATION-CLIENT-AUTH`_ draft 03.
```
```suggestion
      - `OAUTH-ATTESTATION-CLIENT-AUTH`_.
```

peppelinux commented on docs/en/wallet-attestation.rst:
```diff
@@ -293,21 +293,27 @@ using the method `POST <https://datatracker.ietf.org/doc/html/rfc6749#section-3.
     POST /token HTTP/1.1
     Host: wallet-provider.example.org
     Content-Type: application/x-www-form-urlencoded
+    OAuth-Client-Attestation: ICB7DQogICAgImFsZyI6ICJFUzI1NiIsDQogICAgImtpZCI6ICJ2YmVYSmtzTTQ1eHBodEFObkNpRzZtQ3l1VTRqZkdOem9wR3VLdm9nZzljIiwNCiAgICAidHlwIjogIndhcitqd3QiDQogIH0.ew0KICAiaXNzIjogImh0dHBzOi8vd2FsbGV0LXByb3ZpZGVyLmV4YW1wbGUub3JnL2luc3RhbmNlL3ZiZVhKa3NNNDV4cGh0QU5uQ2lHNm1DeXVVNGpmR056b3BHdUt2b2dnOWMiLA0KICAic3ViIjogImh0dHBzOi8vd2FsbGV0LXByb3ZpZGVyLmV4YW1wbGUub3JnLyIsDQogICJjaGFsbGVuZ ...
```
the payload contains `\r\n` and anso the kid in the CNF

this JWT should be recreated 


![immagine](https://github.com/user-attachments/assets/94be7995-3ffc-4517-b32d-9aba7227078e)

peppelinux replied:
```suggestion
    OAuth-Client-Attestation: eyJ0eXAiOiJ3YWxsZXQtYXR0ZXN0YXRpb24rand0IiwiYWxnIjoiRVMyNTYiLCJraWQiOiJFVUcwRXZUVmlJNUVOWkF3VUNJVU1nUEFZOF9VSEluXzJISFpTMTd0X0MwIn0.eyJpc3MiOiAiaHR0cHM6Ly93YWxsZXQtcHJvdmlkZXIuZXhhbXBsZS5vcmcvaW5zdGFuY2UvdmJlWEprc000NXhwaHRBTm5DaUc2bUN5dVU0amZHTnpvcEd1S3ZvZ2c5YyIsICJzdWIiOiAiaHR0cHM6Ly93YWxsZXQtcHJvdmlkZXIuZXhhbXBsZS5vcmcvIiwgImNoYWxsZW5nZSI6ICI2ZWM2OTMyNC02MGE4LTRlNWItYTY5Ny1hNzY2ZDg1NzkwZWEiLCAiaGFyZHdhcmVfc2lnbmF0dXJlIjogIktvWklodmNOQVFjQ29JQXdnQUlCLi4ucmVkYWN0ZWQiLCAiaW50ZWdyaXR5X2Fzc2VydGlvbiI6ICJvMk5tYlhSdllYQndiR1V0WVhCd1lYLi4ucmVkYWN0ZWQiLCAiaGFyZHdhcmVfa2V5X3RhZyI6ICJXUWh5RHltRktzUDk1aUZxcHpkRURXVzRsN2FWbmEyRm40SkNlV0hZdGJVPSIsICJjbmYiOiB7Imp3ayI6IHsiY3J2IjogIlAtMjU2IiwgImt0eSI6ICJFQyIsICJ4IjogIjRITnB0SS14cjJwanlSSktHTW56NFdtZG5RRF91SlNxNFI5NU5qOThiNDQiLCAieSI6ICJMSVpuU0IzOXZGSmhZZ1MzazdqWEU0cjMtQ29HRlF3WnRQQklScXBObHJnIn19LCAidnBfZm9ybWF0c19zdXBwb3J0ZWQiOiB7Imp3dF92Y19qc29uIjogeyJhbGdfdmFsdWVzX3N1cHBvcnRlZCI6IFsiRVMyNTZLIiwgIkVTMzg0Il19LCAiand0X3ZwX2pzb24iOiB7ImFsZ192YWx1ZXNfc3VwcG9ydGVkIjogWyJFUzI1NksiLCAiRWREU0EiXX19LCAiaWF0IjogMTY4NjY0NTExNSwgImV4cCI6IDE2ODY2NTIzMTV9.ClxWia-9QHysYKW3WIEEFAFUIK1-seMHddhpW-2bKHRMqcLvRdck7Pvq0S-9uIaS5N-j7BPDGAJqQCR9A-haLw
```

Commit:
Apply suggestions from code review

Co-authored-by: m-basili <m.basili@ipzs.it>

peppelinux reviewed (commented):
fix: jwt examples

Commit:
fix: jwt examples

Commit:
fix: attestation based client auth http headers parameters

peppelinux commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -17,7 +17,7 @@ The relevant entities and interfaces involved in the issuance flow are:
         - National Identity Provider: It represents preexisting identity systems based on SAML2 or OpenID Connect Core 1.0, already in production in each Member State (eg: the Italian SPID and CIE id schemes notified eIDAS with *LoA* **High**, see `SPID/CIE OpenID Connect Specifications <https://italia.github.io/spid-cie-oidc-docs/en/>`_).
 
-    - *(Q)EAA Provider*: It represents the Issuer of (Q)EAAs. It is composed of:
+    - *(Q)EAA Provider*: It represents the Issuer of (Q)EAAs. It is composed by:
```
```suggestion
    - *(Q)EAA Provider*: It represents the Issuer of (Q)EAAs. It is composed with the following components:
```

peppelinux commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -105,7 +105,7 @@ In this section a *Wallet Initiated Flow* is outlined, where the User receives t
   * MUST create the ``code_verifier`` with enough entropy random string using the unreserved characters with a minimum length of 43 characters and a maximum length of 128 characters, making it impractical for an attacker to guess its value. The value MUST be generated following the recommendation in Section 4.1 of :rfc:`7636`.
   * signs this request using the private key that is created during the setup phase to obtain the Wallet Attestation. The related public key that is attested by the Wallet Provider is provided within the Wallet Attestation ``cnf`` claim.
-  * MUST create the value of the ``client_assertion`` parameter according to OAuth 2.0 Attestation-based Client Authentication [`oauth-attestation-draft <https://vcstuff.github.io/draft-ietf-oauth-attestation-based-client-auth/draft-ietf-oauth-attestation-based-client-auth.html>`_], since in this flow the Pushed Authorization Endpoint is a protected endpoint. The ``client_assertion`` value MUST N ...
+  * MUST the ``OAuth-Client-Attestation`` and  ``OAuth-Client-Attestation-PoP`` parameters according to OAuth 2.0 Attestation-based Client Authentication [`OAUTH-ATTESTATION-CLIENT-AUTH`_], since in this flow the Pushed Authorization Endpoint is a protected endpoint. The first JWT is the Wallet Attestation JWT and the second JWT is the Wallet Attestation Proof of Possession.
```
```suggestion
  * MUST use the ``OAuth-Client-Attestation`` and  ``OAuth-Client-Attestation-PoP`` parameters according to OAuth 2.0 Attestation-based Client Authentication [`OAUTH-ATTESTATION-CLIENT-AUTH`_], since in this flow the Pushed Authorization Endpoint is a protected endpoint.
```

Commit:
Apply suggestions from code review

m-basili committed:
fix: removed kid from wallet attestation pop jwt

m-basili commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -129,11 +129,11 @@ Below a non-normative example of the PAR.
     Host: eaa-provider.example.org
     Content-Type: application/x-www-form-urlencoded
+    OAuth-Client-Attestation: eyJhbGciOiJFUzI1NiIsImtpZCI6IkVVRzBFdlRWaUk1RU5aQXdVQ0lVTWdQQVk4X1VISW5fMkhIWlMxN3RfQzAifQ.eyJpc3MiOiAiaHR0cHM6Ly9jbGllbnQuZXhhbXBsZS5jb20iLCAiYXVkIjogImh0dHBzOi8vYXMuZXhhbXBsZS5jb20iLCAibmJmIjogMTMwMDgxNTc4MCwgImV4cCI6IDEzMDA4MTkzODB9._v3bjJelKI0TNpbc4ysS7yJupwSZzMPQ0ZQ9N5zj8XGQ_T3NN9bghUyVzegR60xokqBnqmMS4iYgPOL7ekEspw
+    OAuth-Client-Attestation-PoP: eyJhbGciOiJFUzI1NiIsImtpZCI6IkVVRzBFdlRWaUk1RU5aQXdVQ0lVTWdQQVk4X1VISW5fMkhIWlMxN3RfQzAifQ.eyJpc3MiOiAiaHR0cHM6Ly9jbGllbnQuZXhhbXBsZS5jb20iLCAiYXVkIjogImh0dHBzOi8vYXMuZXhhbXBsZS5jb20iLCAibmJmIjogMTMwMDgxNTc4MCwgImV4cCI6IDEzMDA4MTkzODB9._v3bjJelKI0TNpbc4ysS7yJupwSZzMPQ0ZQ9N5zj8XGQ_T3NN9bghUyVzegR60xokqBnqmMS4iYgPOL7ekEspw
```
```suggestion
    OAuth-Client-Attestation-PoP: eyJhbGciOiJFUzI1NiJ9.eyJpc3MiOiIgaHR0cHM6Ly9jbGllbnQuZXhhbXBsZS5jb20iLCJhdWQiOiIgaHR0cHM6Ly9hcy5leGFtcGxlLmNvbSIsImp0aSI6IjVlZmY5YzFiLWVkMGQtNDdlOC1hNTUzLWY3NGRmMWJiZWVkZCIsImlhdCI6MTcyMjI0OTQ0NywiZXhwIjoxNzIyMjQ5NzQ3fQ.aZpx7u7R-W8q7fJh9BEaRf8LM7RQRxAVc-okalAVqxHWqUMh3ehYukMLaCsiDQ33pyS41Y5PEsZ3HXwAXQ3nMg
```

peppelinux replied:
`jwt-client-attestation-pop` typ header is unspecified, therefore what would be the reason it get it here?

m-basili replied:
You are right, I removed typ in the jwt header

peppelinux replied:
therefore the original example that I have changed was fine, do you confirm?

m-basili commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -233,19 +233,13 @@ The ``client_assertion`` is signed using the private key that is created during
-        WF0IjoxNTYyMjYyNjE2fQ.2-GxA6T8lP4vfrg8v-FdWP0A0zdrj8igiMLvqRMUvwnQg
-        4PtFLbdLXiOSsX0x7NVY-FNyJK70nfbV37xRZT3Lg
+    OAuth-Client-Attestation: eyJhbGciOiJFUzI1NiIsImtpZCI6IkVVRzBFdlRWaUk1RU5aQXdVQ0lVTWdQQVk4X1VISW5fMkhIWlMxN3RfQzAifQ.eyJpc3MiOiAiaHR0cHM6Ly9jbGllbnQuZXhhbXBsZS5jb20iLCAiYXVkIjogImh0dHBzOi8vYXMuZXhhbXBsZS5jb20iLCAibmJmIjogMTMwMDgxNTc4MCwgImV4cCI6IDEzMDA4MTkzODB9._v3bjJelKI0TNpbc4ysS7yJupwSZzMPQ0ZQ9N5zj8XGQ_T3NN9bghUyVzegR60xokqBnqmMS4iYgPOL7ekEspw
+    OAuth-Client-Attestation-PoP: eyJhbGciOiJFUzI1NiIsImtpZCI6IkVVRzBFdlRWaUk1RU5aQXdVQ0lVTWdQQVk4X1VISW5fMkhIWlMxN3RfQzAifQ.eyJpc3MiOiAiaHR0cHM6Ly9jbGllbnQuZXhhbXBsZS5jb20iLCAiYXVkIjogImh0dHBzOi8vYXMuZXhhbXBsZS5jb20iLCAibmJmIjogMTMwMDgxNTc4MCwgImV4cCI6IDEzMDA4MTkzODB9._v3bjJelKI0TNpbc4ysS7yJupwSZzMPQ0ZQ9N5zj8XGQ_T3NN9bghUyVzegR60xokqBnqmMS4iYgPOL7ekEspw
```
```suggestion
    OAuth-Client-Attestation-PoP: eyJhbGciOiJFUzI1NiJ9.eyJpc3MiOiIgaHR0cHM6Ly9jbGllbnQuZXhhbXBsZS5jb20iLCJhdWQiOiIgaHR0cHM6Ly9hcy5leGFtcGxlLmNvbSIsImp0aSI6IjVlZmY5YzFiLWVkMGQtNDdlOC1hNTUzLWY3NGRmMWJiZWVkZCIsImlhdCI6MTcyMjI0OTQ0NywiZXhwIjoxNzIyMjQ5NzQ3fQ.aZpx7u7R-W8q7fJh9BEaRf8LM7RQRxAVc-okalAVqxHWqUMh3ehYukMLaCsiDQ33pyS41Y5PEsZ3HXwAXQ3nMg
```

peppelinux replied:
`jwt-client-attestation-pop` typ header is unspecified, therefore what would be the reason it get it here?

m-basili replied:
fixed

m-basili reviewed (approved):
fix: update OAuth-Client-Attestation-PoP value in the non-normative example

m-basili commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -461,14 +469,11 @@ The JOSE header of the Wallet Attestation proof of possession MUST contain:
-    * - **kid**
-      -  Unique identifier of the ``jwk`` inside the ``cnf`` claim of Wallet Attestation as base64url-encoded JWK Thumbprint value.
-      - :rfc:`7638#section_3`.
     * - **typ**
       -  It MUST be set to ``jwt-client-attestation-pop``
       -  Currently under discussion in [`oauth-attestation-draft <https://vcstuff.github.io/draft-ietf-oauth-attestation-based-client-auth/draft-ietf-oauth-attestation-based-client-auth.html>`_].
```
```suggestion
```

m-basili replied:
@peppelinux, check please.

Commit:
Apply suggestions from code review

Co-authored-by: m-basili <m.basili@ipzs.it>
